### PR TITLE
Fix ION returning DID Key objects

### PR DESCRIPTION
**Pull request by decentralgabe**
- fix ion returning DID key objects
- refactored DB options
- updated config examples
- now able to run the ssi service without docker again 😄

### CODE REVIEW

File: pkg/service/did/ion.go [hunk 1]
```diff
@@ -257,7 +257,7 @@ func (h *ionHandler) GetDID(ctx context.Context, request GetDIDRequest) (*GetDID
 func (h *ionHandler) GetDIDs(ctx context.Context) (*GetDIDsResponse, error) {
 	logrus.Debug("getting stored did:ion DIDs")
 
-	gotDIDs, err := h.storage.GetDIDs(ctx, did.KeyMethod.String(), new(ionStoredDID))
+	gotDIDs, err := h.storage.GetDIDs(ctx, did.IONMethod.String(), new(ionStoredDID))
```
Comment (decentralgabe) [inline pkg/service/did/ion.go:260]: fix was here, rest is DB refactor

File: config/config.toml [hunk 2]
```diff
@@ -25,11 +25,24 @@ enable_allow_all_cors = true
...
 
-[services.storage_option]
-address = "localhost:6379"
-password = ""
+# example bolt config with filepath option
+storage = "bolt"
+
+[[services.storage_option]]
+id = "boltdb-filepath-option"
+option = "bolt.db"
```
Comment (andresuribe87) [inline config/config.toml:34]: I'm wary of running with bolt, since it might hide some issues of the default way in which we are intending to run the service.

File: config/config.toml [hunk 3]
```diff
@@ -25,11 +25,24 @@ enable_allow_all_cors = true
...
+
+# exmaple redis config with two options
```
Comment (andresuribe87) [inline config/config.toml:36]:
```suggestion
# example redis config with two options
```

File: config/config.toml [hunk 4]
```diff
@@ -38,7 +51,7 @@ password = "default-password"
 
 [services.did]
 name = "did"
-methods = ["key", "web", "ion"]
+methods = ["key", "web"]
```
Comment (andresuribe87) [inline config/config.toml:54]: Curious why this change?

File: pkg/storage/redis.go [hunk 5]
```diff
@@ -60,6 +63,47 @@ func (b *RedisDB) Init(i interface{}) error {
...
+		}
+	}
+	if len(address) == 0 {
+		err = errors.New("redis address must not be empty")
+		return
+	}
+	if len(password) == 0 {
+		err = errors.New("redis password must not be empty")
+		return
+	}
```
Comment (andresuribe87) [inline pkg/storage/redis.go:103]: Seem repeated. Is it possible to DRY?

**Review by andresuribe87 — APPROVED** [review]

File: config/config.toml (hunk 2 above)
Comment (decentralgabe) [reply config/config.toml:34]: this shouldn't matter since compose/CI run with redis

Comment (decentralgabe) [reply config/config.toml:34]: this shouldn't matter since compose/CI run with redis

File: config/config.toml (hunk 4 above)
Comment (decentralgabe) [reply config/config.toml:54]: having ion as a method requires a write node, and until ours is a bit more stable would like to avoid putting it here for everyone to use

File: pkg/storage/redis.go (hunk 5 above)
Comment (decentralgabe) [reply pkg/storage/redis.go:103]: condensed this into a single if. need to check whether the cases were hit for required options, so think this is as simple as it can be

File: config/config.toml (hunk 4 above)
Comment (andresuribe87) [reply config/config.toml:54]: This file is the one used for local development. I don't follow why everyone would use it if we leave this in here. 

I'm also confused by why the values in this field require a write node. For instance "web" doesn't allow any writes, since there is no way or programmatically writing a did:web DID. Can you elaborate?

File: config/config.toml (hunk 2 above)
Comment (andresuribe87) [reply config/config.toml:34]: I think it matters because we're having significant differences between running así service locally, and what's being run in CI. I think we should strive to have the differences be minimal, so that it's easy to locally reproduce problems that happen in CI.

File: config/config.toml (hunk 4 above)
Comment (decentralgabe) [reply config/config.toml:54]: why does local deployment need a dependency on an ion write node?

not all DID methods require similar infra. we could refactor to separate out creating from anchoring...maybe in #340

File: config/config.toml (hunk 2 above)
Comment (decentralgabe) [reply config/config.toml:34]: there are reasons to just run the service in isolation, and to run it in a prod-like environment.

the prod-like environment is using compose. bolt is suitable for running the service in isolation.

we've also taken the approach of introducing a storage abstraction. if you're asserting that our abstraction is faulty then we should fix that -- either by making it better or removing it entirely.

**Comment by decentralgabe** [comment]
@andresuribe87 going to merge, if anything blocking we can re open or discuss more in #340
